### PR TITLE
refactor (genetics parser): changing logic how QTLs processed

### PR DESCRIPTION
**Pull request by DSuveges**
Context: [#3287](https://github.com/opentargets/issues/issues/3287), [#3283](https://github.com/opentargets/issues/issues/3283)

The parser and the data is updated. The data is here:
```
gs://otar000-evidence_input/Genetics_portal/json/genetics-portal-evidence-2024-04-16.json.gz
```

## Benchmark against old data

**Questions:**

- Has the number of evidence changed?
- Has the number of evidence with directionality assignment changed?
- How many evidence have different QTL effect?
- How many times did the directionality changed?

**Answers:**

- There's no change in the number of evidence: 811,749
- Verified that the schema of the two datasets is the same.
- The number of evidence with directionality assessment dropped 12% to 210,050, which means 25.8% of evidence have directionality.
- Then number of evidence, where the directionality assignment changed is 67,170 representing ~31% of all evidence with directionality.

### CODE REVIEW

File: modules/GeneticsPortal.py [hunk 1]
```diff
@@ -448,15 +464,16 @@ def process_variant_rsid(variant_index: str):
...
-def write_evidence_strings(evidence_df: DataFrame, output_file: str) -> int:
-    """
-    Exports the table to a compressed JSON file containing the evidence strings.
-    Pandas is used to export it to a single file, not a directory.
-    """
-    evidence_df.coalesce(1).write.format("json").mode("overwrite").option(
-        "compression", "gzip"
-    ).save(output_file)
-    return 0
+def write_evidence_strings(evidence_df: DataFrame, output_file: str) -> None:
```
[the remark this answers is not in the document]
Comment (DSuveges) [reply modules/GeneticsPortal.py:467]: Yes. This parser has never been integrated into the snakemake pipeline, it runs as a dataproc job, and as such there's no proper environment set up, so cannot be imported from common. It's not elegant, and we need to sort this out. I have plans with the parsers, but the potential gain is not too big.

File: modules/GeneticsPortal.py [hunk 2]
```diff
@@ -3,47 +3,55 @@
...
-    functions as f,
-    Window,
-)
+from pyspark.sql import Column, DataFrame, SparkSession, Window
+from pyspark.sql import functions as f
+from pyspark.sql import types as t
 
-from common.evidence import detect_spark_memory_limit
 
+def detect_spark_memory_limit():
```
[the remark this answers is not in the document]
Comment (DSuveges) [reply modules/GeneticsPortal.py:17]: See previous comment.

File: modules/GeneticsPortal.py [hunk 3]
```diff
@@ -3,47 +3,55 @@
...
+    cause Java heap space errors, even though there is plenty of RAM available. To fix this, we detect the total amount
+    of physical memory and allow Spark to use (almost) all of it."""
+    mem_gib = virtual_memory().total >> 30
+    return int(mem_gib * 0.9)
 
-def get_biggest_effect(df: DataFrame) -> DataFrame:
-    """Filter dataframe for rows with the highest absolute QTL effect.
+
+def get_most_significant_qtl_effect(df: DataFrame) -> DataFrame:
+    """Filter dataframe for rows with the most significant QTL effect.
```
[the remark this answers is not in the document]
Comment (DSuveges) [reply modules/GeneticsPortal.py:26]: Nah. Significance is always implies statistical context.

**Review by ireneisdoomed — APPROVED** [review]
Ty for addressing my questions. All good!

**Pull request merged**